Rockerbox doc (#867)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/rockerbox/setting-up-rockerbox.mdx

</div>

Once you have confirmed that the source platform supports sending events to Rockerbox, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Rockerbox**.
2. Assign a name to your destination and click **Continue**.

## Connection settings

To successfully configure Rockerbox as a destination, configure the following settings:

<img src="../../../assets/event-stream-destinations/rockerbox-connection-settings.png" alt="Rockerbox Connection Settings in RudderStack" />

- **Advertiser Id**: Enter your Rockerbox Advertiser Id.

<div class="infoBlock">
Advertiser Id is a required field for sending events to Rockerbox via <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>. To obtain it, contact your <a href="https://help.rockerbox.com/article/5t050dmcxv-webhooks#requirements:~:text=or%20other%20formats-,Server%2Dside%20specs,-You%20will%20need">Rockerbox account manager</a>.
</div>

> - **Map your RockerBox events to RudderStack Events**: Enter the corresponding RudderStack and RockerBox events you want to be mapped.

Reviewer: ```suggestion
- **Map your RockerBox events to RudderStack Events**: Enter the RudderStack event and the corresponding RockerBox event to be mapped.
```

Reviewer 2: Yes, we can map to multiple events to rockerBox as well as vice versa.

    
### Client-side events filtering

> 

Reviewer: ```suggestion

This setting lets you specify which events should be blocked or allowed to flow through to Rockerbox.

```

Reviewer: Moved this sentence before the image and after the heading, for better readability.

<img src="../../../assets/event-stream-destinations/rockerbox-connection-settings-2.png" alt="Rockerbox Connection Settings in RudderStack" />

This setting lets you specify which events should be blocked or allowed to flow through to Rockerbox.

<div class="infoBlock">
This option is applicable only if you're sending events to Rockerbox via the web device mode. Refer to the <Link to="/sources/event-streams/sdks/event-filtering/">Client-side Event Filtering</Link> guide for more information on this feature.
</div>

### Web SDK settings

<img src="../../../assets/event-stream-destinations/rockerbox-connection-settings-3.png" alt="Rockerbox Connection Settings in RudderStack" />

These settings are applicable only if you are sending events to Rockerbox via the web device mode:

> - **Client Auth Id**: Enter your Rockerbox Client Auth Id.

Reviewer: ```suggestion
- **Client Auth ID**: Enter your Rockerbox Client Auth ID.
```

<div class="infoBlock">

> You can obtain the  Client Auth Id from your Rockerbox All Pages tag available in the Rockerbox Attribution platform. For more information on obtaining it, refer to the <a href="https://help.rockerbox.com/article/unydvsq7jr-site-direct-all-pages-pixel#:~:text=Your%20pixel%20will%20have%20an%20alphanumeric%20ID%20instead%20of%20the%20CLIENT_AUTH_ID%20placeholder">Rockerbox documentation</a>.

Reviewer: ```suggestion
You can obtain the  Client Auth ID from your <strong>Rockerbox All Pages</strong> tag available in the Rockerbox Attribution platform. For more information on obtaining it, refer to the <a href="https://help.rockerbox.com/article/unydvsq7jr-site-direct-all-pages-pixel#:~:text=Your%20pixel%20will%20have%20an%20alphanumeric%20ID%20instead%20of%20the%20CLIENT_AUTH_ID%20placeholder">Rockerbox documentation</a>.
```

Reviewer: @ishachopra0401 highlighted the name of the tag, please confirm.

</div>

- **Enable Cookie Sync**: Enable this setting if you are using a custom domain. It ensures that the Custom Tracking Domain functionality works as expected.
- **Use device-mode to send events**: Enable this setting to send all the events from the JavaScript SDK to Rockerbox via the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link>.
- **Use device-mode to send track events**: Enable this setting to send **only** `track` events from the JavaScript SDK to Rockerbox via the <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link>. Refer to the <Link to="#hybrid-mode">Hybrid mode</Link> for more information.
- **OneTrust Cookie Categories**: This setting lets you map <Link to="/sources/sdks/rudderstack-javascript-sdk/consent-managers/onetrust/">OneTrust</Link> cookie consent groups to RudderStack's consent purposes.

## Hybrid mode

> Hybrid mode lets you use a destination in the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud</Link> as well as <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link> simultaneously. It means that you can route some calls through server side (Cloud Mode), whereas others through the client side SDK (Device Mode). It is very helpful to achieve some use-case scenarios only possible in a particular connection mode.

Reviewer: ```suggestion
Hybrid mode lets you use a destination in the <Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud</Link> as well as <Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link> simultaneously. It means that you can route some calls through server side (<Link to="/destinations/rudderstack-connection-modes/#cloud-mode">cloud mode</Link>), whereas others through the client side SDK (<Link to="/destinations/rudderstack-connection-modes/#device-mode">device mode</Link>). This is helpful when you want to achieve specific use-case scenarios only possible in a particular connection mode.
```

> In RockerBox, the `identify` and `page` calls are supported in device mode whereas the `track` calls are supported in both the cloud and device mode.

Reviewer: We should call this bit out in an info block, IMO. Thoughts?
